### PR TITLE
Switch to gogo for protobufs

### PR DESCRIPTION
**Pull request by mestery**
The broader ligato project is using gogo [1] to generate protobufs, this
patch moves networkservicemesh to using gogo as well.

[1] https://github.com/gogo/protobuf

Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

**Comment by edwarnicke** [comment]
Could you summarize some of the pros and cons here?  I'm curious to learn :)

**Comment by mestery** [comment]
Per discussion in IRC, we'll close this one for now, as gogo and goproto are not compatible it seems, and grpc depends on goproto.
